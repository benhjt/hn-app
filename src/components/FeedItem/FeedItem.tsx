import React from 'react';

import { Link } from 'react-router-dom';

export type FeedItemData = {
  id: number;
  title: string;
  points: number;
  user: string;
  time: number;
  time_ago: string;
  comments_count: number;
  type: string;
  url: string;
  domain: string;
};

type FeedItemProps = {
  item: FeedItemData;
  index: number;
};

export const FeedItem: React.FC<FeedItemProps> = ({ item, index }) => (
  <div>
    <div>{index}</div>
    <div>
      <div>
        <a href={item.url} target="_blank" rel="noopener noreferrer">
          {item.title}
        </a>{' '}
        (
        <a href={item.domain} target="_blank" rel="noopener noreferrer">
          {item.domain}
        </a>
        )
      </div>
      <div>
        {item.points} points by{' '}
        <Link to={`/user/${item.user}`}>{item.user}</Link> {item.time_ago} |{' '}
        <Link to={`/story/${item.id}`}>{item.comments_count} comments</Link>
      </div>
    </div>
  </div>
);

export default FeedItem;
