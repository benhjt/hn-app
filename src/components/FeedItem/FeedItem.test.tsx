import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import FeedItem from './FeedItem';

const item = {
  id: 33934580,
  title: 'This cool story!',
  points: 54,
  user: 'username123',
  content: '<p>test test test</p>',
  time: 1670693712,
  time_ago: '8 hours ago',
  comments_count: 226,
  type: 'link',
  url: 'https://www.google.com',
  domain: 'google.com',
  comments: [],
};

describe('FeedItem', () => {
  it('renders correctly', () => {
    const { container } = render(<FeedItem item={item} index={1} />, {
      wrapper: BrowserRouter,
    });
    expect(container).toMatchSnapshot();
  });
});
