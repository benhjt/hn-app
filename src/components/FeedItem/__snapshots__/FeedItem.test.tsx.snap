// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedItem renders correctly 1`] = `
<div>
  <div
    class="flex"
  >
    <div
      class="flex flex-col"
    >
      <div
        class="flex flex-row align space-x-2"
      >
        <a
          class="peer visited:text-gray-500"
          href="https://www.google.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          This cool story!
        </a>
        <a
          class="peer-visited:text-gray-500"
          href="google.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          (
          google.com
          )
        </a>
      </div>
      <div
        class="flex flex-row text-gray-500 text-xs"
      >
        54
         points by 
        <a
          class="hover:underline"
          href="/user/username123"
        >
          username123
        </a>
         
        8 hours ago
        <div
          class="px-2"
        >
          |
        </div>
        <a
          class="hover:underline"
          href="/story/33934580"
        >
          226
           comments
        </a>
      </div>
    </div>
  </div>
</div>
`;
